src/ledger/LedgerManagerImpl.cpp

```diff
@@ -1550,10 +1563,43 @@ LedgerManagerImpl::applyTransactions(
         }
         ++txNum;
 
-        tx->apply(mApp.getAppConnector(), ltx, tm, mutableTxResult, subSeed);
-        tx->processPostApply(mApp.getAppConnector(), ltx, tm, mutableTxResult);
         TransactionResultPair results;
         results.transactionHash = tx->getContentsHash();
+
+        if (expectedResultsIter)
+        {
+            releaseAssert(*expectedResultsIter !=
+                          expectedResults->results.end());
+            while ((*expectedResultsIter)->transactionHash !=
```

Comment:
This looks a bit suspicious: the order of results should match the order of transactions application exactly. I think you can do `expectedResults->at(i)` instead of the loop.

Reply:
Correct, the order should match, but the loop accounts for the case when there are gaps in the results stored in the archive, which I believe can be the case when there are ledgers with empty txn sets (?)

Reply:
if you have ledgers with empty sets, then `txs` should be empty as well, so we should use the same index logic for both results and txs